### PR TITLE
Don't print join/leave messages in singleplayer (again)

### PR DESCRIPTION
**Pull request by LNJ2**
This was caused by: https://github.com/minetest/minetest/commit/0b0075e6ad0b3110cabdfc92cedb0a24d2b5ec42

### CODE REVIEW

**Comment by SmallJoker** [comment]
I disagree with this pull. You can see your player name on servers, so why not let the people know their name in singleplayer too?

File: builtin/game/misc.lua [hunk 1]
```diff
@@ -99,7 +101,9 @@ core.register_on_leaveplayer(function(player, timed_out)
 	if timed_out then
 		announcement = announcement .. " (timed out)"
 	end
-	core.chat_send_all(announcement)
```
Comment (celeron55) [inline builtin/game/misc.lua]: This conditional should be above "local announcement = " for optimization and to make it clear the announcement isn't being used for anything at all in singleplayer. :+1: then.

File: builtin/game/misc.lua [hunk 2]
```diff
@@ -99,7 +101,9 @@ core.register_on_leaveplayer(function(player, timed_out)
...
+	if not core.is_singleplayer() then
```
Comment (celeron55) [inline builtin/game/misc.lua]: This conditional should be above "local announcement = " for optimization and to make it clear the announcement isn't being used for anything at all in singleplayer. :+1: then.

**Comment by LNJ2** [comment]
You don't need to know who you are in singleplayer, because you're every time the player 'singleplayer' and if you really need to know this use /me.

**Comment by LNJ2** [comment]
Moved conditional above "local announcement = " and squashed.

**Comment by sfan5** [comment]
:+1:

**Comment by SmallJoker** [comment]
Duplicate of #4474

**Comment by LNJ2** [comment]
Ups... sorry

**Comment by est31** [comment]
I'd prefer #4474 over this one, implementation wise, as it doesn't add to vertical bloat. I'm undecided concept wise whether the thing these two PRs want is desireable.

**Comment by paramat** [comment]
:-1: for reasons given in the other PR.

**Comment by nerzhul** [comment]
:-1: let the core side code and don't move this logic to lua

**Comment by paramat** [comment]
Closing due to dissapproval / preference of other PR.

**Pull request closed without merging**